app/controllers/reservation_frames_controller.rb

```diff
@@ -1,7 +1,58 @@
 class ReservationFramesController < ApplicationController
+  before_action :signed_in_planner, only: [:new, :create]
+
   def index
   end
 
   def new
+    a = Time.local(2021, 1, 1, 10)
+    @def_time_frame = (0..15).map{|i| "#{t_format(a + 30.minute * i)}-#{t_format(a + 30.minute * (i+1))}"}
+  end
+
+  def create
+    args = params.require(:reservation_frame)
+    d = Date.parse(args[:reserved_date])
+    t = args[:time_frame].split("-")
+    s = t[0].split(":")
+    e = t[1].split(":")
+
+    @start_at = Time.local(2021, 1, 1, s[0].to_i, s[1].to_i)
+    @end_at = Time.local(2021, 1, 1, e[0].to_i, e[1].to_i)
+    
+    if d.saturday?
+      st = Time.local(2021, 1, 1, 11)
+      en = Time.local(2021, 1, 1, 15)
+      flash[:alert] = {time_frame: "予約日時が範囲外です"} if @start_at < st || @end_at > en
+    end
+
+    time_frame = TimeFrame.new(
+      start_at: @start_at,
+      end_at: @end_at
+    )
+    if time_frame.valid?
+      time_frame.save
+    else
+      flash[:alert] = time_frame.errors.messages
+    end
+
+    reservation_frame = ReservationFrame.new(
+      planner_id: current_user.id,
+      time_frame_id: time_frame.id,
+      reserved_date: d,
+      status: args[:status],
+    )
```

Comment:
引数を間違えると意図しないユーザーで`ReservationFrame`を作成してしまう恐れがあるため、ログインしているユーザーから`ReservationFrame`を作れるようにした方が良いです！
あと記述も減るし！

```suggestion
    reservation_frame = current_user.reservation_frames.build(
      time_frame_id: time_frame.id,
      reserved_date: d,
      status: args[:status],
    )
```

Reply:
こういう書き方 知らなかったので参考になります! ありがとうございます